tests/transforms/test_roi_transforms.py

```diff
@@ -150,54 +150,151 @@ def test_crop_roi_mask(mask, expected, raises_error):
+            ([[1]], motion_border(0, 0, 0, 1), False),
+            (
+                [[0, 0, 0],
+                 [0, 1, 0],
+                 [0, 0, 0]], motion_border(1, 1, 1, 1), True),
+            (
+                [[0, 1, 0],
+                 [0, 1, 0],
+                 [0, 0, 0]], motion_border(1, 1, 0, 1), True),
+            (
+                [[0, 1, 0],
+                 [0, 1, 0],
+                 [0, 0, 0]], motion_border(1, 1, 1, 1), False),
+            (
+                [[0, 0, 0],
+                 [0, 1, 0],
+                 [0, 1, 0]], motion_border(1, 1, 1, 0), True),
+            (
+                [[0, 0, 0],
+                 [0, 1, 0],
+                 [0, 1, 0]], motion_border(1, 1, 1, 1), False),
+            (
+                [[0, 0, 0],
+                 [1, 1, 0],
+                 [0, 0, 0]], motion_border(0, 1, 1, 1), True),
+            (
+                [[0, 0, 0],
+                 [1, 1, 0],
+                 [0, 0, 0]], motion_border(1, 1, 1, 1), False),
+            (
+                [[0, 0, 0],
+                 [0, 1, 1],
+                 [0, 0, 0]], motion_border(1, 0, 1, 1), True),
+            (
+                [[0, 0, 0],
+                 [0, 1, 1],
+                 [0, 0, 0]], motion_border(1, 1, 1, 1), False),
+            (
+                [[0, 0, 0, 0, 0, 0],
+                 [0, 0, 0, 0, 0, 0],
+                 [0, 0, 1, 1, 0, 0],
+                 [0, 0, 1, 1, 0, 0],
+                 [0, 0, 0, 0, 0, 0],
+                 [0, 0, 0, 0, 0, 0]], motion_border(2, 2, 2, 2), True),
+            (
+                [[0, 0, 0, 0, 0, 0],
+                 [0, 0, 0, 0, 0, 0],
+                 [0, 0, 1, 1, 0, 0],
+                 [0, 0, 1, 1, 0, 0],
+                 [0, 0, 0, 0, 0, 0],
+                 [0, 0, 0, 0, 0, 0]], motion_border(2, 3, 2, 2), False),
+                ])
+def test_motion_exclusion(dense_mask, max_correction_vals, expected):
+    coo = coo_matrix(dense_mask)
+    roi = roi_transforms._coo_mask_to_LIMS_compatible_format(coo)
+    roi['max_correction_up'] = max_correction_vals.up
+    roi['max_correction_down'] = max_correction_vals.down
+    roi['max_correction_right'] = max_correction_vals.right
+    roi['max_correction_left'] = max_correction_vals.left
+
+    valid = roi_transforms._motion_exclusion(roi, coo.shape)
+
+    assert valid == expected
+
+
+@pytest.mark.parametrize(
+        "dense_mask, npixel_threshold, expected",
+        [
+            ([[1]], 0, True),
+            ([[1]], 1, False),
+            ([[0, 0, 0],
+              [0, 1, 0],
+              [0, 0, 0]], 4, False),
+            ([[0, 0, 0],
+              [0, 1, 1],
+              [0, 1, 1]], 4, False),
+            ([[0, 0, 0],
+              [1, 1, 1],
+              [0, 1, 1]], 4, True)
+            ])
+def test_small_size_exclusion(dense_mask, npixel_threshold, expected):
+    coo = coo_matrix(dense_mask)
+    roi = roi_transforms._coo_mask_to_LIMS_compatible_format(coo)
```

Comment:
Nit: Nicer if you made a fixture that provided "lims compatible" rois from numpy arrays, since you do this a few times in the test code.

Reply:
Acknowledged. These tests were so fixture heavy when I got to tweaking/replacing them, I found it much harder to figure out what was going on. This seemed pretty light.